js/utils-wc.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Normally "boolean" attribute on HTML elements is when
+ * the attribute is there or not.
+ *
+ * @param {*} input
+ * @returns
+ */
+export const isNotNullOrEmptyString = (input) =>
+  typeof input === 'string' && input !== '' && input !== 'null'
+
+// TODO: FIXME
+export const isValidCustomElement = (classObj) => {
+  const prototypeOf = Object.getPrototypeOf(classObj)
+  return prototypeOf.name === 'HTMLElement'
+}
+
+export const assertIsValidCustomElementName = (
+  elementName /*: string */ = '',
+) /*: assert is ... */ => {
+  if (/^[a-z]([\w\d-])+$/.test(elementName) === false) {
+    const message = `Invalid element name "${elementName}", it must only contain letters and dash.`
+    throw new Error(message)
+  }
+}
```

Comment:
There should be more component "interface" contract check.  Done before registering. Or check in a different window realm. Check things like presence of slot names, attributes. So that if someone wants to override, we can check early if the component's compatible 

TODO